app/src/main/java/com/github/siela1915/bootcamp/UploadingRecipeFragment.java

```diff
@@ -155,6 +180,63 @@ public void onClick(View v) {
         return view;
     }
 
+
+    public void setupIngredientAutocomplete(AutoCompleteTextView ingredientAutoComplete, Map<String, Integer> idMap, IngredientAutocomplete apiService){
+        ingredientAutoComplete.setThreshold(1);
+        //clear the field if the user wants to re-enter the ingredient name
+        ingredientAutoComplete.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if(hasFocus){
+                    ingredientAutoComplete.setText("");
+                }
+            }
```

Comment:
If I understand correctly the code clears the auto-complete field whenever it gains focus, which could be annoying for users who only want to add to the existing text. Instead, it would be better to clear the field only if the existing text is the same as the auto-complete text. :)